rpc/wrtc_signaling_answerer.go

```diff
@@ -334,42 +335,17 @@ func (ans *webrtcSignalingAnswerer) answer(client webrtcpb.SignalingService_Answ
 		}
 	}()
 
-	// only send once since exchange may end or ICE may end
-	var sendDoneErrorOnce sync.Once
-	sendDone := func() error {
-		var err error
-		sendDoneErrorOnce.Do(func() {
-			err = client.Send(&webrtcpb.AnswerResponse{
-				Uuid: uuid,
-				Stage: &webrtcpb.AnswerResponse_Done{
-					Done: &webrtcpb.AnswerResponseDoneStage{},
-				},
-			})
-		})
-		return err
-	}
+	serverChannel := aa.server.NewChannel(pc, dc, aa.hosts)
 
 	var exchangeCtx context.Context
 	var exchangeCancel func()
 	if initStage.Init.Deadline != nil {
-		exchangeCtx, exchangeCancel = context.WithDeadline(ans.closeCtx, initStage.Init.Deadline.AsTime())
+		exchangeCtx, exchangeCancel = context.WithDeadline(aa.closeCtx, initStage.Init.Deadline.AsTime())
 	} else {
-		exchangeCtx, exchangeCancel = context.WithTimeout(ans.closeCtx, getDefaultOfferDeadline())
+		exchangeCtx, exchangeCancel = context.WithTimeout(aa.closeCtx, getDefaultOfferDeadline())
 	}
-
-	errCh := make(chan error)
 	defer exchangeCancel()
-	sendErr := func(err error) {
-		if isEOF(err) {
-			ans.logger.Warnf("answerer swallowing err %v", err)
-			return
-		}
-		ans.logger.Warnf("answerer received err %v of type %T", err, err)
-		select {
-		case <-exchangeCtx.Done():
-		case errCh <- err:
```

Comment:
While I (accidentally?) left `errCh` around, I didn't copy over the functionality and wrote errors into it. `errCh` was (entirely?) used to track problems communicating with the signaling server. I'm asserting those errors aren't fatal. The new code logs these errors, but does not otherwise influence the return values/`successful` value of this function.